### PR TITLE
Compiler warning and error fixes

### PR DESCRIPTION
**Pull request by gacholio**
Fix a few things found while attempting a debug windows build.

[ci skip]

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
FYI to reviewers to check the "Ignore Whitespace" box/button as there were some tabs/spaces changes here. You can also add `?w=1` to the end of the URL.

File: runtime/compiler/runtime/MethodMetaData.c [hunk 1]
```diff
@@ -313,43 +313,42 @@ static void printMapTable(TR_StackMapTable * stackMapTable, U_8 * addressOfFirst
 static void fastwalkDebug(J9TR_MethodMetaData * methodMetaData, UDATA offsetPC, TR_StackMapTable * stackMapTable, TR_MapTableEntry * mapTableEntry)
    {
    if (methodMetaData && (methodMetaData->flags & JIT_METADATA_IS_STUB))
-      {
       printf("Stub Metadata 0x%p, nothing to dump", methodMetaData);
```
Comment (keithc-ca) [inline runtime/compiler/runtime/MethodMetaData.c]: I realize it was this way before, but is the `\n` missing in the format string?

Comment (gacholio) [reply runtime/compiler/runtime/MethodMetaData.c]: I don't disagree, but I was making only the compiler error fix here (locals declared after code in C file).

Comment (gacholio) [reply runtime/compiler/runtime/MethodMetaData.c]: Fixed (and another one).

**Comment by keithc-ca** [comment]
Jenkins compile win32 jdk8

**Pull request merged**